planner/core/errors.go

```diff
@@ -51,6 +51,7 @@ const (
 	codeWrongNumberOfColumnsInSelect = mysql.ErrWrongNumberOfColumnsInSelect
 	codeWrongValueCountOnRow         = mysql.ErrWrongValueCountOnRow
 	codeTablenameNotAllowedHere      = mysql.ErrTablenameNotAllowedHere
+	codePrivilegeCheckFail           = mysql.ErrUnknown
```

Comment:
I think we can use a custom error code like line22-27. 
The custom error code will convert to `ErrUnknow` when returning to MySQL client.

Reply:
If I change the error code, the test-suite will fail.  See #8838, this is a refactor first.  I will change the individual error codes to be MySQL specific after.